### DISCUSSION TITLE
What should we put in the base code?

### DISCUSSION THREAD

**Discussion by sonwan2020** [Azure-Samples/java-microservices-aca-lab#40] [General]
First, the engineers can follow the steps to finish the lab step by step, and there might be some incremental changes during the labs.
For example, when we import the service bus for messaging, in the step by step guide, we add a dependency to the pom.xml in the sample project, [sample pr link](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure-Samples%2Fjava-microservices-aca-lab%2Fpull%2F28%23pullrequestreview-2312557816&data=05%7C02%7CSongbo.Wang%40microsoft.com%7C6e4342ea09a046d520f508dcd7dea041%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638622598426778147%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=vwY2INZBsGUY6YwQEIrU9OqJ1t9HpfKmxwn%2FlwriUAM%3D&reserved=0)

And also we want to offer an automatic azd template for the entire solution, so we should have everything ready in the base code.

In my mind, we should merge the final code into the solution, in the step by step guide, we tell them to check the related dependencies are in the feature related pom files.
In this way, 
* no blocker for the automation
* readers don't have to make code changes

**Comment by sonwan2020** [comment]
@vermegi I see your comments on some of the PRs, what's your suggestions on this topic?

**Reply by vermegi** [reply]
Could we maybe merge the final code to a branch in the repo? That way the final code is available for full deploy through azd, you just pick the right branch for it. And that way the main branch stays clean for learners who want to run through the full lab. 

What do you think? Is that a valid approach?

**Comment by sonwan2020** [comment]
in some 'core' scenarios, new branches might be an option
it will be hard to maintain the branches in the future

**Reply by vermegi** [reply]
well, in case there are changes in the main branch, we need to make sure to merge these also to the different branches. 
I don't think we will end up with that many of them, so might be still doable.

**Comment by vermegi** [comment]
What we could do is, branches for: 
- passwordless
- ai
- messaging

I think those are the 3 main scenario's that have code changes. 

We could then also build container images with the same tag off of each of these branches and put them in the mcr repo to be consumed by the azd/automation lab. 

What do you think of this idea?
